src/cpp/rtps/reader/CompoundReaderListener.cpp

```diff
@@ -44,6 +44,7 @@ bool CompoundReaderListener::hasReaderAttached(){
 	return false;
 }
 ReaderListener* CompoundReaderListener::getAttachedListener(){
+	std::lock_guard<std::mutex> lock(attached_listener_mutex);
```

Comment:
Why would the read access need to be mutexed? The caller might still get a wrong answer since the value might change immediately after the mutex is unlocked before the caller receives the value.